lib/reverse_words.rb

```diff
@@ -1,4 +1,50 @@
 # A method to reverse each word in a sentence, in place.
+
 def reverse_words(my_words)
-  raise NotImplementedError
+  # raise NotImplementedError
+  first_index = 0
+  last_index = 0
+
+  return nil if my_words == nil
+
+  total_indices = my_words.length - 1
+
+  while my_words[last_index] != nil
+
+    while last_index < total_indices
+      if my_words[last_index + 1] == " "
+        break
+      else
+        last_index += 1
+      end
+    end
+
+    my_words = reverse(my_words, first_index, last_index)
```

Comment:
Since you're manipulating the string in place, you don't need to assign the return value of reverse to my_words.